static/src/javascripts/projects/commercial/modules/hosted/video.js

```diff
@@ -116,7 +116,8 @@ define([
 
                         player.ready(function () {
                             var vol;
-                            duration = parseInt(this.duration(), 10);
+                            var player = this;
```

Comment:
It is needed here, this is what was broken. In this scope, `player` points to the wrong object (always the last video), so `this` is the correct reference.